docs/uml-diagrams/ListCommand.puml

```diff
@@ -0,0 +1,145 @@
+    -hasQuantity:boolean
+    -hasPrice:boolean
+    -hasExpiry:boolean
+    -firstParam:String
+    -secondParam:String
+    -sortBy:String
+    -isReverse:boolean
+    +ListCommand(hasQuantity:boolean, hasPrice:boolean, hasExpiry:boolean,
+    firstParam:String, secondParam:String, sortBy:String, isReverse:boolean)
+    +execute():void
+    +isExit():boolean
+}
+
+class Item {
+    {static}+sortByName():Comparator<Item>
+    {static}+sortByQuantity():Comparator<Item>
+    {static}+sortByPrice():Comparator<Item>
+    {static}+sortByExpiry():Comparator<Item>
+}
+
+class Inventory {
+    {static}+getItems():ArrayList<Item>
+}
+
+class Ui {
+     {static}+listIntro(size:int):void
+    {static}+listItem(item:Item, index:int, hasQuantity:boolean, hasPrice:boolean,
+    {static}hasExpiry:boolean, firstParam:String, secondParam:String):void
+}
+
+class Parser {
+    {static}-parseListCommand(input:String):ListCommand
+}
+@enduml
+
+@startuml
+hide footbox
+participant "SuperTracker" as SuperTracker <<class>> #f5e3a9
+participant ":ListCommand" as ListCommand #cbf7f4
+participant "Inventory" as Inventory <<class>> #d5eac2
+participant "Item" as Item <<class>> #fbffb2
+participant "Collections" as Collections <<class>> #ffcdd6
+participant "Ui" as Ui <<class>> #e5c2ea
+participant ":ArrayList<Item>" as ArrayList #bcf7cf
+
+SuperTracker -> ListCommand : execute()
+activate ListCommand #cbf7f4
+
+ListCommand -> Inventory : getItems()
+activate Inventory #d5eac2
+
+Inventory -> ArrayList ** : new ArrayList<>(items:Collection<Item>)
+activate ArrayList #bcf7cf
+ArrayList --> Inventory : items:ArrayList<Item>
+deactivate ArrayList
+
+Inventory --> ListCommand : items:ArrayList<Item>
+deactivate Inventory
+
+ListCommand -> Ui : listIntro(size:int)
+activate Ui #e5c2ea
+Ui --> ListCommand
+deactivate Ui
+
+alt sortBy == "q"
+
+ListCommand -> Item : sortByQuantity()
+activate Item #fbffb2
+Item --> ListCommand : comparator:Comparator<Item>
+deactivate Item
+
+else sortBy == "p"
+
+ListCommand -> Item : sortByPrice()
+activate Item #fbffb2
+Item --> ListCommand : comparator:Comparator<Item>
+deactivate Item
+
+else sortBy == "e"
+
+ListCommand -> Item : sortByExpiry()
+activate Item #fbffb2
+Item --> ListCommand : comparator:Comparator<Item>
+deactivate Item
+
+
+else else
+
+ListCommand -> Item : sortByName()
+activate Item #fbffb2
+Item --> ListCommand : comparator:Comparator<Item>
+deactivate Item
+
+
+
+end
+
+ListCommand -> ArrayList : sort(comparator:Comparator<Item>)
+activate ArrayList #bcf7cf
+ArrayList --> ListCommand
+deactivate ArrayList
+
+opt isReverse
+
+ListCommand -> Collections : reverse(items:ArrayList<Item>)
+activate Collections #ffcdd6
+Collections --> ListCommand
+deactivate Collections
+
+end
+
+loop items
+
+ListCommand -> Ui : listItem(item:Item, index:int, hasQuantity:boolean, hasPrice:boolean, hasExpiry:boolean, firstParam:String, secondParam:String):void)
+activate Ui #e5c2ea
+Ui --> ListCommand
+deactivate Ui
+
+end
```

Comment:
In all your sequence diagram, only objects are interacting. So should you remove the <<`class`>> decorator above?